cmd/snap-recovery/partition/partition.go

```diff
@@ -0,0 +1,29 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package partition
+
+import (
+	"github.com/snapcore/snapd/gadget"
+)
+
+// PartitionTool is used to query and manipulate the device partition table.
+type partitionTool interface {
+	DeviceInfo() (*gadget.LaidOutVolume, error)
+	CreatePartitions(*gadget.LaidOutVolume, []bool) (map[string]string, error)
```

Comment:
s/CreatePartions/Create/ ?

should this one be idempotent? what happens if the partions are already there

the []bool param is a bit opaque/unusual

Reply:
I added a "TODO" about the []bool - I agree this is not ideal and it should probably be worked out by the tool itself.

Reply:
I would like to work on the unusedPartitions in a followup maybe? I think this will require some more work.

Reply:
Fwiw, I plan to work on this once https://github.com/snapcore/snapd/commit/aeb381468a439ca55e9431ffe96f4e45d8f00348 is in - this is an integration test that runs the real sfdisk so that we have a more realistic test for when sfdisk works out what is availalbe and what is not.